core/lib/spree/event/execution.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Spree
+  module Event
+    # Execution of a {Spree::Event::Listener}
+    #
+    # When an event is fired, it executes all subscribed listeners. Every single
+    # execution is represented as an instance of this class. It contains the
+    # result value of the listener, along with helpful metadata as the time of
+    # the execution or a benchmark for it.
+    #
+    # You'll most likely interact with this class for debugging or logging
+    # purposes through the returned value in {Spree::Event.fire}.
+    class Execution
+      # The listener to which the execution belongs
+      #
+      # @return [Spree::Event::Listener]
+      attr_reader :listener
+
+      # The value returned by the {#listener}'s block
+      #
+      # @return [Any]
+      attr_reader :result
+
+      # Benchmark for the {#listener}'s block
+      #
+      # @return [Benchmark::Tms]
+      attr_reader :benchmark
+
+      # Time of execution in UTC standard
+      #
+      # @return [Time]
+      attr_reader :utc_time
+
+      # @private
+      def initialize(listener:, result:, benchmark:)
+        @listener = listener
+        @result = result
+        @benchmark = benchmark
+        @utc_time = Time.now.utc
```

Comment:
This is a nit, but I'd probably pass this from the outside. Looks like `Execution` is mostly a value object, and it might be useful to pass the time from outside for testing purposes?

Reply:
Yes, I agree. I already had thought about it. I changed it, but I left `Time.now.utc` as the argument's default value. In a way, it makes sense to have it optionally tied to the initialization method. What do you think?

I also changed `Spree::Event::Event` to be consistent.